de.metas.serviceprovider/src/main/java/de/metas/serviceprovider/everhour/EverhourImporterService.java

```diff
@@ -0,0 +1,284 @@
+import de.metas.serviceprovider.timebooking.importer.failed.FailedTimeBookingRepository;
+import de.metas.user.UserId;
+import de.metas.util.Check;
+import de.metas.util.Loggables;
+import de.metas.util.time.LocalDateInterval;
+import lombok.NonNull;
+import org.adempiere.ad.trx.api.ITrxManager;
+import org.adempiere.exceptions.AdempiereException;
+import org.slf4j.Logger;
+import org.springframework.stereotype.Service;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.ZoneOffset;
+import java.util.List;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.regex.Matcher;
+
+import static de.metas.issue.tracking.everhour.api.EverhourConstants.TaskIdSource.GITHUB_ID;
+import static de.metas.serviceprovider.everhour.EverhourImportConstants.PROCESSING_DATE_INTERVAL_SIZE;
+import static de.metas.serviceprovider.timebooking.importer.ImportConstants.IMPORT_TIME_BOOKINGS_LOG_MESSAGE_PREFIX;
+
+
+@Service
+public class EverhourImporterService implements TimeBookingsImporter
+{
+	private static final Logger log = LogManager.getLogger(EverhourImporterService.class);
+
+	private final ReentrantLock lock = new ReentrantLock();
+
+	private final EverhourClient everhourClient;
+	private final ExternalReferenceRepository externalReferenceRepository;
+	private final ImportQueue<ImportTimeBookingInfo> timeBookingImportQueue;
+	private final FailedTimeBookingRepository failedTimeBookingRepository;
+	private final ObjectMapper objectMapper;
+	private final ITrxManager iTrxManager;
+
+	public EverhourImporterService(final EverhourClient everhourClient, final ExternalReferenceRepository externalReferenceRepository, final ImportQueue<ImportTimeBookingInfo> timeBookingImportQueue, final FailedTimeBookingRepository failedTimeBookingRepository, final ObjectMapper objectMapper, final ITrxManager iTrxManager)
+	{
+		this.everhourClient = everhourClient;
+		this.externalReferenceRepository = externalReferenceRepository;
+		this.timeBookingImportQueue = timeBookingImportQueue;
+		this.failedTimeBookingRepository = failedTimeBookingRepository;
+		this.objectMapper = objectMapper;
+		this.iTrxManager = iTrxManager;
+	}
+
+	public void importTimeBookings(@NonNull final ImportTimeBookingsRequest request)
+	{
+		try
+		{
+			acquireLock();
+
+			importFailedTimeBookings();
+
+			final LocalDateInterval localDateInterval = LocalDateInterval.of(request.getStartDate(), request.getEndDate());
+
+			localDateInterval.divideUsingStep(PROCESSING_DATE_INTERVAL_SIZE)
+					.stream()
+					.map(interval -> buildGetTeamTimeRecordsRequest(request.getAuthToken(), interval))
+					.map(everhourClient::getTeamTimeRecords)
+					.flatMap(List::stream)
+					.filter(timeRecord -> isGithubID(timeRecord.getTask().getId()))
+					.forEach(this::importTimeBooking);
+		}
+		catch (final Exception e)
+		{
+			Loggables.withLogger(log, Level.ERROR).addLog(IMPORT_TIME_BOOKINGS_LOG_MESSAGE_PREFIX + e.getMessage());
```

Comment:
i think in such a case it's good log log, but AFAIU (didn't look deep) you should rethrow while adding currently available info
e.g. 
```java
throw AdempiereException.wrapIfNeeded(e).appendParametersToMessage().setParameter("request", request);
```